htmresearch/frameworks/layers/l2_l4_network_creation.py

```diff
@@ -303,12 +303,21 @@ def createMultipleL4L2Columns(network, networkConfig):
   for i in range(networkConfig["numCorticalColumns"]):
     suffixSrc = "_" + str(i)
     for j in range(networkConfig["numCorticalColumns"]):
-      if i != j:
-        suffixDest = "_" + str(j)
+      suffixDest = "_" + str(j)
+      if i < j:
+        # use delayed link for lateral connections because c_i is computed
+        # before c_j
         network.link(
-            "L2Column" + suffixSrc, "L2Column" + suffixDest,
-            "UniformLink", "",
-            srcOutput="feedForwardOutput", destInput="lateralInput")
+          "L2Column" + suffixSrc, "L2Column" + suffixDest,
+          "UniformLink", "",
+          srcOutput="feedForwardOutput", destInput="lateralInput",
+          propagationDelay=1)
+      elif i > j:
+        network.link(
+          "L2Column" + suffixSrc, "L2Column" + suffixDest,
+          "UniformLink", "",
+          srcOutput="feedForwardOutput", destInput="lateralInput",
+          propagationDelay=0)
```

Comment:
Per discussion with @vitaly-krugl  and @scottpurdy, the current implementation is sensitive to the order of the columns. With the current implementation, we need to specify the order of L2 columns and adjust the delay accordingly. 

@vitaly-krugl will change the network class to make it insensitive of orders of regions with the same phase, and we will need to update the logic here after that change.

I have specified the phase of L2 columns to ensure that they will be called in the desired order.